src/ts/settlement.ts

```diff
@@ -12,6 +12,31 @@ import {
 } from "./order";
 import { TypedDataDomain } from "./types/ethers";
 
+/**
+ * The stage an interaction should be executed in.
+ */
+export enum InteractionStage {
+  /**
+   * The interaction will be executed before any trading occur.
+   *
+   * This can be used, for example, to perform as EIP-2612 `permit` call for a
+   * user trading in the current settlement.
+   */
+  PRE = 0,
+  /**
+   * The interaction will be executed after all trade sell amounts are
+   * transferred into the settlement contract, but before the buy amounts are
+   * transferred out to the traders.
+   *
+   * This can be used, for example, to interact with on-chain AMMs.
+   */
+  INTER = 1,
```

Comment:
Not super happy with this name... Open to suggestions.

Reply:
`MID`?

Reply:
https://english.stackexchange.com/questions/71986/if-pre-is-previous-post-is-after-what-is-current suggests per.

Otherwise maybe INTRA rather than INTER?

Reply:
- peri- To me this means around more than in (like perimeter of a circle for example).
- intra- As in "intra-settlement" sounds good to me as well. I had originally taken inter- to mean "inter-trade-exectutions" but I like intra- a little better.
- mid- As in "mid-settlement" also works. Personally, intra- sounds a tiny bit better, but I'm really on the fence about this.